backend/dataall/modules/s3_datasets/cdk/dataset_stack.py

```diff
@@ -437,20 +449,68 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
             },
         )
 
+        glue_sec_conf_enc_key = kms.Key(
```

Comment:
I want to call out that this key comes with key policy which includes the block of code that delegates permissiosn to IAM, such as 

```
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::ACCOUNT_ID:root"
            },
            "Action": "kms:*",
            "Resource": "*"
        },
```

meaning along with the principals who have access in the key policy - a role/user with kms permissions via IAM policy will be able to use this key as well

For the sake of this key - which encrypts Glue Job and Crawler I do not see a major risk but if there is a need to change can set the `policy=` parameter directly in `kms.Key(...)` rather than using `add_to_resource_policy`

Reply:
i put the CDK admin for now, should i add this policy too Noah?